File: go-client-sdk/pkg/models/operations/getallcrmemails.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetAllCrmEmailsQueryParams struct {
	AccessToken string   `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool    `queryParam:"style=form,explode=true,name=allFields"`
	Cursor      *string  `queryParam:"style=form,explode=true,name=cursor"`
	Limit       *float64 `queryParam:"style=form,explode=true,name=limit"`
}

type GetAllCrmEmailsResponseBody struct {
	Emails         []shared.Email `json:"emails,omitempty"`
	NextPageCursor *string        `json:"nextPageCursor,omitempty"`
}

type GetAllCrmEmailsRequest struct {
	QueryParams GetAllCrmEmailsQueryParams
}

type GetAllCrmEmailsResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetAllCrmEmailsResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getallcrmemails.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetAllCrmEmailsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;
}


export class GetAllCrmEmailsResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=emails", elemType: shared.Email })
  emails?: shared.Email[];

  @SpeakeasyMetadata({ data: "json, name=nextPageCursor" })
  nextPageCursor?: string;
}


export class GetAllCrmEmailsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetAllCrmEmailsQueryParams;
}


export class GetAllCrmEmailsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetAllCrmEmailsResponseBody;
}
